aci/resource_aci_cloudtemplateextnetwork.go

```diff
@@ -0,0 +1,447 @@
+package aci
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/ciscoecosystem/aci-go-client/v2/client"
+	"github.com/ciscoecosystem/aci-go-client/v2/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAciCloudTemplateforExternalNetwork() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAciCloudTemplateforExternalNetworkCreate,
+		UpdateContext: resourceAciCloudTemplateforExternalNetworkUpdate,
+		ReadContext:   resourceAciCloudTemplateforExternalNetworkRead,
+		DeleteContext: resourceAciCloudTemplateforExternalNetworkDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceAciCloudTemplateforExternalNetworkImport,
+		},
+
+		SchemaVersion: 1,
+
+		CustomizeDiff: func(_ context.Context, diff *schema.ResourceDiff, v interface{}) error {
```

Comment:
what does this customizeDiff do?

Reply:
It validates if appropriate values are set for cloud_vendor parameter for aws, gcp and azure.